dspace-api/src/main/java/org/dspace/app/sitemap/GenerateSitemaps.java

```diff
@@ -220,18 +220,20 @@ public static void generateSitemaps(boolean makeHTMLMap,
             String url = handleURLStem + i.getHandle();
             Date lastMod = i.getLastModified();
 
-            if (makeHTMLMap)
-            {
-                html.addURL(url, lastMod);
-            }
-            if (makeSitemapOrg)
-            {
-                sitemapsOrg.addURL(url, lastMod);
+            if( i.isDiscoverable() ){
+                if (makeHTMLMap)
+                {
+                    html.addURL(url, lastMod);
+                }
+                if (makeSitemapOrg)
+                {
+                    sitemapsOrg.addURL(url, lastMod);
+                }
+
+                c.uncacheEntity(i);
+
+                itemCount++;
             }
-
-            c.uncacheEntity(i);
-
-            itemCount++;
```

Comment:
I'm wondering about this change. It is logical to only increment the item count when we have a discoverable item, but shouldn't we *always* uncache items when we're done with them, whether they are discoverable or not? I suppose this would introduce a performance penalty or even a memory leak?